client/client.go

```diff
@@ -147,9 +168,9 @@ func (client *Client) do(method, path string, query url.Values, headers map[stri
 // response payload into the given value.
 func (client *Client) doSync(method, path string, query url.Values, headers map[string]string, body io.Reader, v interface{}) (*ResultInfo, error) {
 	var rsp response
-
-	if err := client.do(method, path, query, headers, body, &rsp); err != nil {
-		return nil, fmt.Errorf("cannot communicate with server: %s", err)
+	err := client.do(method, path, query, headers, body, &rsp)
```

Comment:
Sorry, it was a reverted change (I started the loop here).

I'll put it back as it was.